ASTProcessor/src/main/java/edu/usb/argos/ASTProcessor/reader/directory_readers/SourceTreeAnalyzer.java

```diff
@@ -0,0 +1,37 @@
+package edu.usb.argos.ASTProcessor.reader.directory_readers;
+
+import edu.usb.argos.ASTProcessor.reader.interfaces.IPathValidator;
+import org.springframework.stereotype.Component;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+@Component
+public class SourceTreeAnalyzer {
+
+    private final IPathValidator pathValidator;
+
+    public SourceTreeAnalyzer(IPathValidator pathValidator) {
+        this.pathValidator = pathValidator;
+    }
+
+    public List<Path> getJavaFiles(Path directoryPath) throws Exception {
+        String javaExtension = ".java";
+        return getFilesByExtension(directoryPath, javaExtension);
+    }
+
+    private List<Path> getFilesByExtension(Path directoryPath, String extension) throws Exception {
```

Comment:
please do not just throw an exception please catch the exception and add a logger please use https://www.slf4j.org/. Additionally it is necessary to catch the specific exception

Reply:
done